docs/packages/dara-components/plotting/bokeh_events.md

```diff
@@ -114,46 +125,62 @@ def interactive_wheat_scatter_plot():
         code="""
             const index = cb_data.source.inspected.indices[0];
             return cb_data.source.data['wheat_class'][index];
-        """
+        """,
     )
 
     # adding the CusomJS object to a TapTool object (to trigger the action whenever the plot is clicked on)
-    taptool = TapTool(callback=click_event(), behavior='inspect')
+    taptool = TapTool(callback=click_event(), behavior="inspect")
     p.add_tools(taptool)
 
-    return Bokeh(p, events=[
-        ('CLICK', [UpdateVariable(lambda ctx: ctx.inputs.new, selected_variety_var)])
-    ])
+    return Bokeh(
+        p,
+        events=[
+            (
+                "CLICK",
+                [selected_variety_var.sync()],
+            )
+        ],
+    )
+
+
+@action
+async def navigate_to(ctx: action.Ctx, selected_variety_var):
+    await ctx.navigate(
+        url=f"https://en.wikipedia.org/wiki/Triticum_{selected_variety_var}",
+        new_tab=True,
+    )
+
 
 @py_component
 def selected_variety_information(selected_variety):
     if selected_variety is None:
-        return  Text(
-            'Click on (inspect) a point on the graph to see it\'s wheat variety class.',
-            align='center'
+        return Text(
+            "Click on (inspect) a point on the graph to see it's wheat variety class.",
+            align="center",
         )
     else:
         return Stack(
-            Text(f'The selected point is of the {selected_variety} wheat variety.'),
+            Text(f"The selected point is of the {selected_variety} wheat variety."),
             Button(
-                f'Read More on Triticum {selected_variety}',
-                onclick=NavigateTo(url=f'https://en.wikipedia.org/wiki/Triticum_{selected_variety}', new_tab=True),
+                f"Read More on Triticum {selected_variety}",
+                onclick=navigate_to(selected_variety),
```

Comment:
This example can use a static url, e.g. `NavigateTo(f'https://(...){selected_variety}')`, no need to make a function